src/Nethermind/Nethermind.Evm/VirtualMachine.cs

```diff
@@ -360,7 +360,6 @@ public TransactionSubstate Run<TTracingActions>(EvmState state, IWorldState worl
                         {
                             _state.InsertCode(callCodeOwner, callResult.Output, spec);
                             currentState.GasAvailable -= codeDepositGasCost;
-                            currentState.CreateList.Add(callCodeOwner);
```

Comment:
Was it too early? Too late? Not on the same EvmState? I see this was added on #4704 so this is just moving this within context of new functionality?

Reply:
it was too late

Reply:
It was too late, needed to add the contract address to the list before initcode starts running, so in case initcode `selfDestruct` is called. the contract is destroyed.